packages/trader/src/Modules/Trading/Components/Form/WheelPicker/wheel-picker.tsx

```diff
@@ -0,0 +1,134 @@
+    }),
+};
+
+type WheelPickerType = {
+    onClick: (id: string) => Promise<void> | void;
+    options: string[];
+    defaultValue?: string;
+    currency?: string;
+};
+
+const WheelPicker = ({ options, onClick, defaultValue, currency }: WheelPickerType) => {
+    const [selectedIndex, setSelectedIndex] = useState(Math.floor(options.length / 2));
+    const [direction, setDirection] = useState('down');
+
+    useEffect(() => {
+        if (defaultValue) {
+            setSelectedIndex(options.indexOf(defaultValue));
+        }
+    }, [defaultValue, options]);
+
+    const handleIndexChange = (newIndex: number, newDirection: 'up' | 'down') => {
+        setDirection(newDirection);
+        setSelectedIndex(newIndex);
+        onClick(options[newIndex]);
+    };
+
+    const handleIncrease = () => {
+        if (selectedIndex > 0) {
+            handleIndexChange(selectedIndex - 1, 'up');
+        }
+    };
+
+    const handleDecrease = () => {
+        if (selectedIndex < options.length - 1) {
+            handleIndexChange(selectedIndex + 1, 'down');
+        }
+    };
+
+    const visibleValues = () => {
+        const values = [options[selectedIndex - 1] || '', options[selectedIndex], options[selectedIndex + 1] || ''];
+        return values;
+    };
+    const hasUndefinedValues = visibleValues().some(option => option === undefined);
+
+    return (
+        <div className='wheel-picker'>
+            <div className='wheel-picker__wheel' key={selectedIndex}>
+                {options.length > 0 &&
+                    !hasUndefinedValues &&
+                    visibleValues().map((value: string | number, index: number) => (
+                        <motion.div
+                            key={index}
+                            variants={variants}
+                            custom={direction}
+                            initial='enter'
+                            animate='center'
+                            exit='exit'
+                        >
+                            <Text
+                                size={index === 1 ? 'xs' : 'xxs'}
+                                line_height={index === 1 ? 'l' : 'm'}
+                                weight={index === 1 ? 'bolder' : 'bold'}
+                                color={index === 1 ? 'default' : 'disabled-1'}
+                                align='center'
+                                as='p'
+                                className={index === 1 ? '' : 'wheel-picker__wheel__placeholder'}
+                            >
+                                {value} {value !== '' ? currency : ''}
+                            </Text>
+                        </motion.div>
+                    ))}
+            </div>
+            <div className='wheel-picker__actions'>
+                <Button
+                    disabled={selectedIndex === 0}
+                    small
+                    className='wheel-picker__actions__btn'
+                    data-testid='up-btn'
```

Comment:
Up to you, but we have a kind of agreement that data-testid names should start with 'dt' , like:
`data-testid='dt_up_btn'`